pkg/occlient/occlient.go

```diff
@@ -2739,7 +2739,7 @@ func (c *Client) CopyFile(localPath string, targetPodName string, targetPath str
 
 	// cmdArr will run inside container
 	cmdArr := []string{"tar", "xf", "-", "-C", targetPath, "--strip", "1"}
-	err := c.ExecCMDInContainer(targetPodName, cmdArr, writer, writer, reader, false)
+	err := c.ExecCMDInContainer(targetPodName, cmdArr, bytes.NewBuffer([]byte{}), bytes.NewBuffer([]byte{}), reader, false)
```

Comment:
can we move the statement in line 2729 instead to may be outside the go func that is proably to the line after 2726
?

Reply:
@anmolbabu You can't do that. If you move defer outside of the go routine into `CopyFile` you will end up in an infinite loop. 
`reader` and `writer` are connected with Pipe. If you move defer to the `CopyFile` it will close the writer side only when `CopyFile` function is finished, and this function will finish only after `ExecCMDInContainer` is done. `ExecCMDInContainer` reads from the reader side of the Pipe, and it will finish only after the writer side is closed. In this case it would be never :-)

Reply:
What @dgolovin did is correct. `writer` shouldn't be used inside  `ExecCMDInContainer`.

Instead if passing empty buffer, we can just use `nil`, `ExecCMDInContainer` will handle that correctly.

Reply:
Good catch @dgolovin

Reply:
@kadel `nil` was also working, but I used [this line 206](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cp/cp.go#L206) as an example. should I update it to `nil`?

Reply:
@kadel have no idea why  `login test` is failing though.

Reply:
> @kadel `nil` was also working, but I used [this line 206](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cp/cp.go#L206) as an example. should I update it to `nil`?

They allocate a new buffer because it is used later on (it is returned from that function).
We never read that output, it is safe to use `nil` here, to clearly mark that that output is discarded. 


> @kadel have no idea why `login test` is failing though.

I see all test green now. I tried `odo login` and works. Maybe it was time timeout in tests :-(

Reply:
I've updated PR to use `nil` instead of empty buffers.